### PR TITLE
Make kube-proxy iptables sync period configurable

### PR DESCRIPTION
**Pull request by thockin**
Useful for testing so manual changes don't get blown away.

### CODE REVIEW

**Comment by thockin** [comment]
@BenTheElder @ArtfulCoder

**Comment by BenTheElder** [comment]
LGTM

File: cmd/kube-proxy/app/server.go [hunk 1]
```diff
@@ -86,6 +87,7 @@ func (s *ProxyServer) AddFlags(fs *pflag.FlagSet) {
 	fs.Var(&s.PortRange, "proxy-port-range", "Range of host ports (beginPort-endPort, inclusive) that may be consumed in order to proxy service traffic. If unspecified (0-0) then ports will be randomly chosen.")
 	fs.StringVar(&s.HostnameOverride, "hostname-override", s.HostnameOverride, "If non-empty, will use this string as identification instead of the actual hostname.")
 	fs.BoolVar(&s.ForceUserspaceProxy, "legacy-userspace-proxy", true, "Use the legacy userspace proxy (instead of the pure iptables proxy).")
+	fs.DurationVar(&s.SyncPeriod, "iptables-sync-period", 5*time.Second, "How often iptables rules are refreshed.")
```
Comment (ArtfulCoder) [inline cmd/kube-proxy/app/server.go]: can you mention some examples for the SyncPeriod in the description.
"e.g. 300ms", "-1.5h" or "2h45m"

Comment (ArtfulCoder) [inline cmd/kube-proxy/app/server.go]: can you mention some examples for the SyncPeriod in the description.
"e.g. 300ms", "1.5h" or "2h45m"
Do we need to say that it should be greater than 0 ? 
For 0 value, the code will panic on NewTicker call

**Comment by ArtfulCoder** [comment]
proxier_test may not compile. you might need to update that.

File: cmd/kube-proxy/app/server.go (hunk 1 above)
Comment (thockin) [inline cmd/kube-proxy/app/server.go]: done in next push.  In general we are very bad about validating flags.  Sigh.

**Comment by thockin** [comment]
New push is up - should pass tests now.

**Comment by ArtfulCoder** [comment]
LGTM
(shippable is still running, but looks good)

**Comment by brendandburns** [comment]
@k8s-bot test this [testing build queue, sorry for the noise]

**Comment by brendandburns** [comment]
Automatic merge from SubmitQueue

**Comment by thockin** [comment]
@k8s-oncall test - please ACK that multiple oncall lists got this.

**Comment by zmerlynn** [comment]
Google secondary ACK

**Comment by zmerlynn** [comment]
Heh. @k8s-oncall also serves as a way to CC myself. I wasn't expecting to see my own comment. :P

**Comment by saad-ali** [comment]
Primary ACK. But with the shuffle looks like I'm no longer primary

**Pull request merged**